### PR TITLE
added jv-hw-injector solution

### PR DESCRIPTION
**Pull request by Basilius-Third**

### CODE REVIEW

File: src/main/java/mate.academy/lib/Injector.java [hunk 1]
```diff
@@ -1,13 +1,77 @@
...
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Field[] declaredField = clazz.getDeclaredFields();
+        Object clazzImplementationInstance = null;
+        for (Field field : declaredField) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                clazzImplementationInstance = createNewInstance(clazz);
```
Comment (PashaTerletskyi) [inline src/main/java/mate.academy/lib/Injector.java:30]: inizialize it in line 26

**Review by PashaTerletskyi — APPROVED** [review]

**Review by testYurka — APPROVED** [review]
good job!!

File: src/main/java/mate.academy/lib/Component.java [hunk 2]
```diff
@@ -1,5 +1,9 @@
 package mate.academy.lib;
```
Comment (KseniiaMakarova) [inline src/main/java/mate.academy/lib/Component.java:1]: change method main() as well please

File: src/main/java/mate.academy/lib/Injector.java [hunk 3]
```diff
@@ -1,13 +1,73 @@
...
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't initialize field value. "
+                            + "Class: " + clazz.getName()
+                            + ". Field: " + field.getName());
+                }
+            }
+        }
+        if (clazzImplementationInstance == null) {
+            clazzImplementationInstance = createNewInstance(clazz);
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/mate.academy/lib/Injector.java:41]: can we remove this check?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/mate.academy/lib/Injector.java (hunk 3 above)
Comment (Basilius-Third) [reply src/main/java/mate.academy/lib/Injector.java:41]: yes we can

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**